packages/react/src/components/ComboBox/ComboBox.tsx

```diff
@@ -37,21 +58,36 @@ const {
   changeInput,
 } = Downshift.stateChangeTypes;
 
-const defaultItemToString = (item) => {
+const defaultItemToString = <ItemType,>(item: ItemType | null) => {
   if (typeof item === 'string') {
     return item;
   }
-
-  return item && item.label;
+  if (typeof item === 'number') {
+    return `${item}`;
+  }
+  if (
+    item !== null &&
+    typeof item === 'object' &&
+    'label' in item &&
+    typeof item['label'] === 'string'
+  ) {
+    return item['label'];
+  }
+  return '';
```

Comment:
It was previously never possible for this function to return an empty string with the nullish coalescing `&&`. It would return `item` if `item.label` was nullish.

```suggestion
  return item;
```

Reply:
but how did it assure that the returned value is a string? maybe we can return
 ```
`${item}`
```

Reply:
but anyways, the code for this is identical as the one used in both `MultiSelect.tsx` and `Dropdown.tsx`